src/fdb5/database/Key.h

```diff
@@ -42,19 +42,25 @@ class Rule;
 
 //----------------------------------------------------------------------------------------------------------------------
 
-class Key {
+class BaseKey {
 
 public: // methods
 
-    explicit Key(const std::shared_ptr<TypesRegistry> reg = nullptr, bool canonical = false);
-    explicit Key(eckit::Stream &, const std::shared_ptr<TypesRegistry> reg = nullptr);
-    explicit Key(const std::string &keys, const Rule* rule);
-    explicit Key(const eckit::StringDict &keys, const std::shared_ptr<TypesRegistry> reg=nullptr);
-    Key(std::initializer_list<std::pair<const std::string, std::string>>, const std::shared_ptr<TypesRegistry> reg=nullptr);
+    BaseKey() : keys_(), names_() {}
+    BaseKey(const BaseKey &key) : keys_(key.keys_), names_(key.names_) {}
 
```

Comment:
redundant, could be:
    BaseKey() = default;
    BaseKey(const BaseKey &key) = default;